app/questionnaire/questionnaire_schema.py

```diff
@@ -1220,6 +1231,15 @@ def _get_calculated_summary_section_dependencies(
 
         return section_dependencies
 
+    def _get_section_ids_for_answer_ids(self, answer_ids: set[str]) -> OrderedSet[str]:
+        section_dependencies: set[str] = set()
+        for answer_id in answer_ids:
+            block = self.get_block_for_answer_id(answer_id)
+            # Type ignore: block_id and section_id is never None
+            section_id = self.get_section_id_for_block_id(block["id"])  # type: ignore
+            section_dependencies.add(section_id)  # type: ignore
+        return OrderedSet(section_dependencies)
```

Comment:
Needs to be an order set on initialisation or you lose the order even if you return `OrderedSet` at the end.
```suggestion
        section_dependencies: set[str] = OrderedSet()
        for answer_id in answer_ids:
            block = self.get_block_for_answer_id(answer_id)
            # Type ignore: block_id and section_id is never None
            section_id = self.get_section_id_for_block_id(block["id"])  # type: ignore
            section_dependencies.add(section_id)  # type: ignore
        return section_dependencies
```